typings/index.d.ts

```diff
@@ -611,10 +625,20 @@ export class CommandInteractionOptionResolver {
   ): NonNullable<CommandInteractionOption['member' | 'role' | 'user']> | null;
   public getMessage(name: string, required: true): NonNullable<CommandInteractionOption['message']>;
   public getMessage(name: string, required?: boolean): NonNullable<CommandInteractionOption['message']> | null;
+  public getFocused<
+    T extends InteractionOptionResolverReturn<Type, AutocompleteInteraction, ApplicationCommandOptionChoice>,
+  >(getFull: true): T;
+  public getFocused<T extends InteractionOptionResolverReturn<Type, AutocompleteInteraction, string | number>>(
+    getFull?: boolean,
+  ): T;
```

Comment:
> What I'm saying is the required option is a hack for typings purposes here. There's no real reason to actually use it. And yes that applies to context menu interactions too! We should probably be using type generics instead

@ckohen Would this type generic work out for what you were suggesting or do you have a better suggestion for this?

Reply:
Yes! this works great The functions themselves don't need generics though, just move them like so (not sure why the diff looks so weird here).

```suggestion
 public getFocused(
   getFull: true
 ): InteractionOptionResolverReturn<Type, AutocompleteInteraction, ApplicationCommandOptionChoice>;
 public getFocused(
    getFull?: boolean,
  ): InteractionOptionResolverReturn<Type, AutocompleteInteraction, string | number>;
```